packages/anoncreds/src/services/tails/BasicTailsFileService.ts

```diff
@@ -0,0 +1,92 @@
+import type { TailsFileService } from './TailsFileService'
+import type { AnonCredsRevocationRegistryDefinition } from '../../models'
+import type { AgentContext, FileSystem } from '@aries-framework/core'
+
+import { AriesFrameworkError, InjectionSymbols, TypedArrayEncoder } from '@aries-framework/core'
+
+export class BasicTailsFileService implements TailsFileService {
+  private tailsDirectoryPath?: string
+
+  public constructor(options?: { tailsDirectoryPath?: string; tailsServerBaseUrl?: string }) {
+    this.tailsDirectoryPath = options?.tailsDirectoryPath
+  }
+
+  public async getTailsBasePath(agentContext: AgentContext) {
+    const fileSystem = agentContext.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    const basePath = `${this.tailsDirectoryPath ?? fileSystem.cachePath}/anoncreds/tails`
+    if (!(await fileSystem.exists(basePath))) {
+      await fileSystem.createDirectory(`${basePath}/file`)
+    }
+    return basePath
+  }
+
+  public async getTailsFilePath(agentContext: AgentContext, tailsHash: string) {
+    return `${await this.getTailsBasePath(agentContext)}/${tailsHash}`
+  }
+
+  public async tailsFileExists(agentContext: AgentContext, tailsHash: string): Promise<boolean> {
+    const fileSystem = agentContext.dependencyManager.resolve<FileSystem>(InjectionSymbols.FileSystem)
+    const tailsFilePath = await this.getTailsFilePath(agentContext, tailsHash)
+    return await fileSystem.exists(tailsFilePath)
+  }
```

Comment:
I think we shouldn't assume the tails file is only accessed using it's hash. Do you think we could also require the revocation registry id etc.. for the methods?

Reply:
This is true but it is hard to reference local tails files in a different way given that we don't know the revocation registry id upfront.

That being said, I think some of those methods are not at the interface level, so for the moment I left only three public methods:
- uploadTailsFile
- getTailsFile
- getTailsBasePath 

The latter is used to indicate anoncreds library where to store the local tails file it will generate. I'll need to re-think this a little bit because ideally it could store in any temporary directory and then the `TailsFileService` can store it in its internal cache once it uploads it.